### ISSUE TITLE
Retrieving minimum and maximum TIME_PERIOD

### ISSUE THREAD

**Issue by edkryx** [sdmx-twg/sdmx-rest#199]
Is there a way to retrieve minimum and maximum TIME_PERIOD values from SDMX Rest API?

I want to retrieve all data from API in Python, however if I load whole table into memory, app will crash. So I want to use parameter startPeriod and endPeriod to fetch data for a year at a time. However, I would need to know from which date data is available. 

Is there a built-in solution for this by any chance?

**Comment by sosna** [comment]
@agent96: Could this be covered by the availability query? In addition to the metrics, we could add the temporal coverage?

**Comment by dosse** [comment]
Indeed, the standard already defines the time range in the availability response, e.g.:  
```xml
<structure:CubeRegion include="true">
    <common:KeyValue id="TIME_PERIOD">
        <common:TimeRange>
            <common:StartPeriod isInclusive="true">1960-01-01T00:00:00</common:StartPeriod>
            <common:EndPeriod isInclusive="true">2022-12-31T23:59:59</common:EndPeriod>
        </common:TimeRange>
    </common:KeyValue>
</structure:CubeRegion>
```

Example implementation: https://sdmx.oecd.org/public/rest/v2/availability/dataflow/OECD.ELS.HD/DSD_HEALTH_STAT@DF_LE/*/*.A.LFEXP.Y.Y0._T._Z._Z._Z._Z._Z._Z._Z

@edkryx Please have a look at the availability documentation: https://github.com/sdmx-twg/sdmx-rest/blob/master/doc/availability.md

**Comment by edkryx** [comment]
I imagine this depends on the source as well. As I am looking at ECB ICB table, there is no such a field as StartPeriod.

https://data-api.ecb.europa.eu/service/dataflow/ECB/ICB/latest?references=all

**Comment by sosna** [comment]
Hi @edkryx. Indeed, implementers don't have to implement the full spec, so it can very well be that a particular service does not support that feature. However, I notice that you used a dataflow query, while @dosse was mentioning an availability query. If the ECB service supports that query, maybe they indicate temporal coverage too. 

In any case, as far as the SDMX-REST spec is concerned, there is a way to support the requirement, so I'll keep this issue closed.
